src/components/ColorPalette/ColorPalette.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react';
+import { colorList } from 'constants';
+import "./ColorPalette.css";
+const ColorPalette = () => {
+    const [colorListStyle, setColorListStyle] = useState("color-list-none")
+    const toggleColorListStyle = () => {
+        setColorListStyle(classDetails => classDetails === "color-list flex-row" ? setColorListStyle("color-list-none") : ("color-list flex-row"))
+    }
+
+    return (
+        <div className='color-palette-action'>
+            <button className='color-selector-btn' onClick={toggleColorListStyle}>
+                <span className="material-icons-outlined app-icon">
+                    palette
+                </span>
+            </button>
+            <div className={colorListStyle}>
+                {
+                    colorList.map((everyColor, index) => {
+                        return (
+                            <span className='color-element' key={index} style={{ backgroundColor: everyColor }} value={everyColor}></span>
```

Comment:
Avoid inline styling 
Instead create diff classes in css
And then match everyColor with that class
```ClassName={curr color?"style class":"default"}```

Reply:
Made the changes related to it using the index property